Strange behaviour using KeyValue store

Hi,

I realise there wont be much information here initially, however I'll put what I have until I can get a reproducible test case.

nats-server -DV
```
# nats-server -DV
[21] 2022/11/14 23:21:59.234003 [INF] Starting nats-server
[21] 2022/11/14 23:21:59.234045 [INF]   Version:  2.9.6
[21] 2022/11/14 23:21:59.234048 [INF]   Git:      [289a9e1]
[21] 2022/11/14 23:21:59.234054 [DBG]   Go build: go1.19.3
```

I have a 5 server cluster running via docker, server configs are as follows:
```
port: 4222
http_port: 8222
server_name: nats-x
client_advertise: 10.x.xx.20
cluster {
  name: "dnCluster"
  port: 6222
  cluster_advertise: 10.x.xx.20:6222
  routes = [
    nats-route://10.x.xx.20:6222
    nats-route://10.x.xx.20:6222
    nats-route://10.x.xx.20:6222
    nats-route://10.x.xx.20:6222
  ]
}
accounts {
 $SYS { users = [ { user: "admin", pass: "xxxx" } ] }
}
jetstream {
  store_dir: /data
  max_memory_store: 268435456
}
```

Creating the KeyValue stores using the go client then php clients can use it via grpc:
TTL is various but generally is around 4 hours
```
s.JetStream.CreateKeyValue(&nats.KeyValueConfig{
		Bucket:      bucket,
		Description: "Session Storage",
		History:     1,
		TTL:         time.Duration(ttl),
		Storage:     nats.FileStorage,
		Replicas:    3,
	})
```

The nats server is generally doing around 5000msg/sec however this issue seems to happen at any time.

Issue 1: 
Random server starts displaying this in the logs even though all servers are up, once this happens these keyvalue stores are non-functioning. Deleting them via cli has no effect ( if it even lets you delete them ). If you manage to delete them you can no longer create a new KeyValue store with the same name.

```
[1] 2022/11/14 06:23:24.196626 [WRN] JetStream cluster stream '$G > KV_SESSION_e0b58a5b-c238-45d3-8ada-837d9bfdeea9' has NO quorum, stalled
[1] 2022/11/14 06:23:27.610976 [WRN] JetStream cluster stream '$G > KV_SESSION_e64684f5-591a-44f3-9478-ed73cfcb233f' has NO quorum, stalled
[1] 2022/11/14 06:23:28.135742 [WRN] JetStream cluster stream '$G > KV_SESSION_02e19464-2a6c-465c-b250-36de31fd643e' has NO quorum, stalled
```

Issue 2:
random segfault:
```
[1] 2022/11/14 09:16:43.271775 [INF] Self is new JetStream cluster metadata leader

2022-11-14T09:41:45.350514792Z
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x3b2ca8]
 
goroutine 20 [running]:
github.com/nats-io/nats-server/v2/server.(*Server).jsStreamInfoRequest(0x0?, 0x0?, 0x40000b0000, 0x0?, {0x40012713b0, 0x43}, {0x40027c6c90, 0x11}, {0x4002524400, 0x7e, ...})
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:1786 +0x958
github.com/nats-io/nats-server/v2/server.(*jetStream).apiDispatch(0x40000f8000, 0x40000de480, 0x40000b0000, 0x4000112d80, {0x40012713b0, 0x43}, {0x40027c6c90, 0x11}, {0x4002524400, 0x7e, ...})
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:764 +0x1cc
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0x40000b0000, 0x0, 0x40000de480, 0x40027c6c78?, {0x4001271180, 0x43, 0x50}, {0x40027c6c78, 0x11, 0x18}, ...)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:3194 +0xa54
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0x40000b0000, 0x4000112d80, 0x40015ae810, {0x4002524400, 0x80, 0x80}, {0x0, 0x0, 0x5d7200?}, {0x4001271180, ...}, ...)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:4235 +0x74c
github.com/nats-io/nats-server/v2/server.(*client).processInboundClientMsg(0x40000b0000, {0x4002524400, 0x80, 0x80})
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:3660 +0x9d4
github.com/nats-io/nats-server/v2/server.(*Server).internalSendLoop(0x400019e000, 0x0?)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/events.go:428 +0xd60
created by github.com/nats-io/nats-server/v2/server.(*Server).setSystemAccount
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:1278 +0x308

[1] 2022/11/14 09:17:14.865616 [INF] 10.x.xx.20:6222(nats-e) - rid:6 - Router connection closed: Read Error

node-e:

[1] 2022/11/14 09:16:43.272941 [INF] JetStream cluster new metadata leader: nats-b/dnCluster

2022-11-14T09:17:14.863325825Z
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x3b2ca8]
goroutine 13 [running]:
github.com/nats-io/nats-server/v2/server.(*Server).jsStreamInfoRequest(0x0?, 0x0?, 0x4000197980, 0x400018e6f0?, {0x4000818aa0, 0x43}, {0x4000b922b8, 0x11}, {0x4000b4ad80, 0x7e, ...})
/home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:1786 +0x958
github.com/nats-io/nats-server/v2/server.(*Server).processJSAPIRoutedRequests(0x400011e000)
/home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:796 +0x27c
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
/home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:3077 +0xbc
```

The issue still happens if I stop all nodes, remove data and restart them all ( as if booting a new cluster )

These 2 things may be related and unfortunately I don't have the docker logs from the previous time it happened to check for segfaults.

In the mean time I'm trying to create a reproducible environment locally.

--- a/server/jetstream_api.go
+++ b/server/jetstream_api.go
@@ -1780,7 +1780,10 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 			if cc.meta != nil {
 				ourID = cc.meta.ID()
 			}
-			bail := !rg.isMember(ourID)
+			// We have seen cases where rg or rg.node is nil at this point,
+			// so check explicitly on those conditions and bail if that is
+			// the case.
+			bail := rg == nil || rg.node == nil || !rg.isMember(ourID)
 			if !bail {
 				// We know we are a member here, if this group is new and we are preferred allow us to answer.
 				bail = rg.Preferred != ourID || time.Since(rg.node.Created()) > lostQuorumIntervalDefault

--- a/server/jetstream_cluster_3_test.go
+++ b/server/jetstream_cluster_3_test.go
@@ -1373,3 +1373,42 @@ func TestJetStreamClusterInterestStreamConsumer(t *testing.T) {
 		t.Fatalf("Should not have any messages left: %d of %d", si.State.Msgs, n)
 	}
 }
+
+func TestJetStreamClusterNoPanicOnStreamInfoWhenNoLeaderYet(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	defer c.shutdown()
+
+	nc := natsConnect(t, c.randomServer().ClientURL())
+	defer nc.Close()
+
+	js, _ := nc.JetStream(nats.MaxWait(500 * time.Millisecond))
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	ch := make(chan struct{})
+	go func() {
+		defer wg.Done()
+
+		for {
+			js.StreamInfo("TEST")
+			select {
+			case <-ch:
+				return
+			case <-time.After(15 * time.Millisecond):
+			}
+		}
+	}()
+
+	time.Sleep(250 * time.Millisecond)
+
+	// Don't care if this succeeds or not (could get a context deadline
+	// due to the low MaxWait() when creating the context).
+	js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+		Replicas: 3,
+	})
+
+	close(ch)
+	wg.Wait()
+}

